packages/core/contracts/UnirepSocial.sol

```diff
@@ -298,6 +301,9 @@ contract UnirepSocial is zkSNARKHelper {
         require(unirep.globalStateTreeRoots(epoch, publicSignals[0]), "Unirep Social: GST root does not exist in epoch");
         require(publicSignals[5] == toEpochKey, "Unirep Social: must prove non-ownership of epk");
 
+        // save proof nullifier
+        usedProofNullifier[proofNullifier] = true;
```

Comment:
Why keep it so far down? You can put this on line 293 so this logic is all together